src/unitquaternion.jl

```diff
@@ -328,6 +328,14 @@ function rotation_between(from::SVector{3}, to::SVector{3})
     @inbounds return QuatRotation(w, v[1], v[2], v[3]) # relies on normalization in constructor
 end
 
+"""
+    slerp(R1::Rotaion{3}, R2::Rotaion{3}, t::Real)
+
+Perform spherical linear interpolation (Slerp) between rotations `R1` and `R2`.
+"""
+Quaternions.slerp(q1::QuatRotation, q2::QuatRotation, t::Real) = QuatRotation(slerp(q1.q, q2.q, t))
+Quaternions.slerp(r1::Rotation{3}, r2::Rotation{3}, t::Real) = typeof(r1)(slerp(QuatRotation(r1), QuatRotation(r2), t))
```

Comment:
I think the default output type should be `QuatRotation` to support partial rotation types such as `RotXY`.
```julia
julia> slerp(RotXY(1,2), RotYZ(4,5), 0)
ERROR: Cannot construct a two-axis rotation from a matrix
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] RotXY
   @ ~/.julia/dev/Rotations/src/euler_types.jl:230 [inlined]
 [3] StaticArray
   @ ~/.julia/dev/StaticArrays/src/convert.jl:167 [inlined]
 [4] slerp(r1::RotXY{Float64}, r2::RotYZ{Float64}, t::Int64)
   @ Rotations ~/.julia/dev/Rotations/src/unitquaternion.jl:337
 [5] top-level scope
   @ REPL[58]:1
```

```suggestion
Quaternions.slerp(r1::Rotation{3}, r2::Rotation{3}, t::Real) = slerp(QuatRotation(r1), QuatRotation(r2), t)
```

>I don't know which representation it should return for arguments with mixed types. The proposed one returns `typeof(r1)` for `slerp(r1, r2, t)`.

I think it would be hard to determine the output types. For example, we would like to have

* `slerp(::RotMatrix{3}, ::RotMatrix{3}, ::Real)` should be `RotMatrix{3}`.
* `slerp(::MRP, ::MRP, ::Real)` should be `MRP`.

But I am not sure what type we need for the following operations. (Related: https://github.com/JuliaGeometry/Rotations.jl/issues/253)

* `slerp(::RotXYZ, ::RotXYZ)`
  * Converting types to `RotXYZ` takes additional computational cost, so returning `QuatRotation` is better for performance.
  * Of course, returning `RotXYZ` is better for consistency.
* `slerp(::MRP, ::RotXYZ)`
  * The output type should be `MRP` or `QuatRotation`.
  * But the current return type of `*(::MRP, ::RotXYZ)` is `RotMatrix{3}`. (x-ref: https://github.com/JuliaGeometry/Rotations.jl/issues/253)

So, it seems better to have `QuatRotation` as a return type of `slerp` for now. We can fix this issue in another PR.